python/cuda_parallel/tests/test_scan_api.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+
+def test_scan_max():
+    # example-begin scan-max
+    import cupy as cp
+    import numpy as np
+
+    import cuda.parallel.experimental.algorithms as algorithms
+
+    def max_op(a, b):
+        return max(a, b)
+
+    h_init = np.array([1], dtype="int32")
+    d_input = cp.array([-5, 0, 2, -3, 2, 4, 0, -1, 2, 8], dtype="int32")
+    d_output = cp.empty_like(d_input, dtype="int32")
```

Comment:
Unless you'd like to be explicit about the `dtype`, this could be replaced with shorter `d_output = cp.empty_like(d_input)`